ui/src/org/pentaho/di/ui/spoon/Spoon.java

```diff
@@ -5363,14 +5365,23 @@ public boolean saveFileAs() throws KettleException {
   public boolean saveFileAs( EngineMetaInterface meta ) throws KettleException {
     boolean saved;
 
-    if ( log.isBasic() ) {
-      log.logBasic( BaseMessages.getString( PKG, "Spoon.Log.SaveAs" ) ); // "Save as..."
+    if ( getLog().isBasic() ) {
+      getLog().logBasic( BaseMessages.getString( PKG, "Spoon.Log.SaveAs" ) ); // "Save as..."
+    }
+
+    if ( meta instanceof TransMeta ) {
+      ( (TransMeta) meta ).setRepository( rep );
+      ( (TransMeta) meta ).setMetaStore( metaStore );
+    } else if ( meta instanceof JobMeta ) {
+      ( (JobMeta) meta ).setRepository( rep );
+      ( (JobMeta) meta ).setMetaStore( metaStore );
```

Comment:
Why the need for the instanceof checks above?  Aren't the .setRepo and .setMetaStore methods defined in the parent class of TransMeta and JobMeta?